icees_api/features/sql.py

```diff
@@ -547,16 +547,27 @@ def create_cohort_view(conn, table_name, cohort_features):
                 "feature_qualifier": {
                     "operator": value["operator"],
                     "value": value["value"],
+                } if "value" in value else {
```

Comment:
@maximusunc Agreed, but this "feature_qualifier" key is a dict item in a list, and this list is created with list comprehension in one statement. If I move the if else statement outside, it will break the list comprehension and this one line statement has to be written without the list comprehension which means append() needs to be used to create the whole list. I guess it is a trade-off between readability and performance. I think with this one else addition, although readability suffers a bit, but not to a point to justify replacing list comprehension yet. So I think it is OK at this point to leave it as is for now, but I definitely hear you. If more logics need to be added in, this definitely needs to be simplified. Hope this justifies this.

Reply:
I see. I think this could still be cleaner though. I would do something like this:
```python
cohort_features = [
    {
        "feature_name": key,
        "feature_qualifier": {
            "operator": value["operator"],
            # value only has two possible keys, value key for one value, and values key for a list of values
            # format value["values"] list into ("value1", "value2", ...,) for SQL IN operator query
            "value": value.get("value") or '("{}")'.format('\", \"'.join(value["values"])),
        }
    }
    for key, value in cohort_features.items()
]

```
It tries to get() the `value` key and then falls back to `values`

Reply:
@maximusunc Many thanks for the cleaner code suggestion. It works well. I also tried the following variant to your suggestion: ```"value": value.get("value", '("{}")'.format('\", \"'.join(value["values"])))```, but got the key error with ```values``` which is a bit puzzling since it should only try to get ```values``` key when getting ```value``` key fails.

Reply:
Hmm interesting! I actually think your variation is more correct.
For my solution:
- if `value` is not defined, it will try and use `values`
- If `value` is defined but falsey, I think it would still try and use `values`, which is not correct
- If `value` is truthy, it'll use it
For your solution:
- If `value` is not defined, it will try and use `values`
- If `value` is defined, it should use it regardless of truthiness.

I would try and debug your solution more to figure out what's going on.

Reply:
@maximusunc Agreed. Thanks for the thorough comparison. I'll look into it further to see why I got values key error with the variation version.

Reply:
This should work:
`"value": value.get("value") if value.get("value") != None else '("{}")'.format('\", \"'.join(value["values"])),`

Reply:
Yes, this does work in SQL. I can see the confusion here, though. To break this down a bit, ```value["values"]``` is a list of strings, say, ```["str1", "str2"]```,  ```'\", \"'.join(value["values"]``` would output a joined string like ```str1", "str2``` which is then further formatted with ```("{}")```to add a starting and ending quotation marks to make the joined string as ```("str1", "str2")``` which becomes a valid SQL IN statement. Hope this clarifies.